pkg/skaffold/runner/dev.go

```diff
@@ -34,30 +34,30 @@ import (
 var ErrorConfigurationChanged = errors.New("configuration changed")
 
 func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer) error {
-	actionPerformed := false
+	if r.changeSet.needsReload {
+		return ErrorConfigurationChanged
```

Comment:
I think we want the logs muted during this time right? otherwise we'll have deployment logs interleaved with the build/deploy logs from the new skaffold runner.

also, `r.changeSet.resetReload()`?

Reply:
In fact, the logs will be stopped when this function returns: [here](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/runner/dev.go#L116).
Also, a new runner, with a new changeset will be created.

Reply:
We should write tests expressing these expectations. 
It seems that there was no log muting in the current version - but I think I agree that interleaving might happen - are you suggesting adding log muting?

Reply:
ahh very nice, thanks for clarifying. this is much cleaner! 👍